sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusSenderAsyncClient.java

```diff
@@ -261,9 +347,16 @@ public String getEntityPath() {
                     parentContext.set(tracerProvider.startSpan(
                         entityContext.addData(HOST_NAME_KEY, link.getHostname()), ProcessKind.SEND));
                 }
+
+                TransactionalState deliveryState = null;
+                if (transactionContext != null && transactionContext.getTransactionId() != null) {
+                    deliveryState = new TransactionalState();
+                    deliveryState.setTxnId(new Binary(transactionContext.getTransactionId().array()));
+                }
```

Comment:
In the case of a transaction, why not just return the link.send(message, deliveryState); and when there isn't one, keep the link.send(message);? Not a fan of accepting null as a normal deliveryState. It would keep the behaviour we want explicit.